### PR TITLE
Add endpoint to get objecttypen of zaakobjecten

### PR DESCRIPTION
**Pull request by ivo-ritense**

### CODE REVIEW

File: document/src/main/java/com/ritense/document/autoconfigure/DocumentAutoConfiguration.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package com.ritense.document.autoconfigure;
 
-import com.ritense.document.config.SpringContextHelper;
+import com.ritense.document.config.DocumentSpringContextHelper;
```
Comment (marijnritense) [inline document/src/main/java/com/ritense/document/autoconfigure/DocumentAutoConfiguration.java:19]: I guess the branch was out of date with development?

File: plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt [hunk 2]
```diff
@@ -228,12 +230,24 @@ class PluginService(
...
+
+    fun createInstance(pluginConfiguration: PluginConfiguration): Any {
         return  pluginFactories.first {
-            it.canCreate(configuration)
-        }.create(configuration)!!
+            it.canCreate(pluginConfiguration)
+        }.create(pluginConfiguration)!!
+    }
+
+    fun <T> createInstanceConditional(clazz: Class<T>, filter: (JsonNode) -> Boolean): T? {
```
Comment (marijnritense) [inline plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt:244]: Maybe `createInstance(clazz: Class<T>, configurationFilter: (JsonNode) -> Boolean)`?

File: plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt [hunk 3]
```diff
@@ -165,12 +167,12 @@ class PluginService(
...
     }
 
-    fun invoke(execution: DelegateExecution, processLink: PluginProcessLink) {
+    fun invoke(execution: DelegateExecution, processLink: PluginProcessLink): Any? {
         val instance = createInstance(processLink.pluginConfigurationId)
 
         val method = getActionMethod(instance, processLink)
         val methodArguments = resolveMethodArguments(method, execution, processLink.actionProperties)
-        method.invoke(instance, *methodArguments)
+        return method.invoke(instance, *methodArguments)
```
Comment (theo-ritense) [inline plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt:175]: Do we have a use case (other than testing) for returning a value on an action? We can't use the returned the way an action is configured right now, can we?

File: plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt [hunk 4]
```diff
@@ -282,6 +296,14 @@ class PluginService(
         }
     }
 
+    fun findPluginConfiguration(pluginDefinitionKey: String, filter: (JsonNode) -> Boolean): PluginConfiguration? {
```
Comment (theo-ritense) [inline plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt:299]: having the createInstanceConditional, this method could be private.

File: plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt (hunk 3 above)
Comment (marijnritense) [reply plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt:175]: When invoking an action through code. So not as important for the development team, but it's useful for implementation teams.

**Review by theo-ritense — APPROVED** [review]

**Review by marijnritense — APPROVED** [review]

**Pull request merged**